sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -0,0 +1,324 @@
+            extract a label from `X` if it is a dataframe, otherwise an empty
+            string is used.
+
+        ax : Matplotlib axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is
+            created.
+
+        **kwargs : dict
+            Additional keyword arguments to be passed to the
+            `plot_method`.
+
+        Returns
+        -------
+        display : :class:`~sklearn.inspection.DecisionBoundaryDisplay`
+            Object that stores the result.
+
+        See Also
+        --------
+        DecisionBoundaryDisplay : Decision boundary visualization.
+        ConfusionMatrixDisplay.from_estimator : Plot the confusion matrix
+            given an estimator, the data, and the label.
+        ConfusionMatrixDisplay.from_predictions : Plot the confusion matrix
+            given the true and predicted labels.
+
+        Examples
+        --------
+        >>> import matplotlib.pyplot as plt
+        >>> from sklearn.datasets import load_iris
+        >>> from sklearn.linear_model import LogisticRegression
+        >>> from sklearn.inspection import DecisionBoundaryDisplay
+        >>> iris = load_iris()
+        >>> X = iris.data[:, :2]
+        >>> classifier = LogisticRegression().fit(X, iris.target)
+        >>> disp = DecisionBoundaryDisplay.from_estimator(
+        ...     classifier, X, response_method="predict",
+        ...     xlabel=iris.feature_names[0], ylabel=iris.feature_names[1],
+        ...     alpha=0.5,
+        ... )
+        >>> disp.ax_.scatter(X[:, 0], X[:, 1], c=iris.target, edgecolor="k")
+        <...>
+        >>> plt.show()
+        """
+        check_matplotlib_support(f"{cls.__name__}.from_estimator")
+
+        if not grid_resolution > 1:
+            raise ValueError(
+                "grid_resolution must be greater than 1. Got"
+                f" {grid_resolution} instead."
+            )
+
+        if not eps >= 0:
+            raise ValueError(
+                f"eps must be greater than or equal to 0. Got {eps} instead."
+            )
+
+        possible_plot_methods = ("contourf", "contour", "pcolormesh")
+        if plot_method not in possible_plot_methods:
+            avaliable_methods = ", ".join(possible_plot_methods)
+            raise ValueError(
+                f"plot_method must be one of {avaliable_methods}. "
+                f"Got {plot_method} instead."
+            )
+
+        x0, x1 = _safe_indexing(X, 0, axis=1), _safe_indexing(X, 1, axis=1)
+
+        x0_min, x0_max = x0.min() - eps, x0.max() + eps
+        x1_min, x1_max = x1.min() - eps, x1.max() + eps
+
+        xx0, xx1 = np.meshgrid(
+            np.linspace(x0_min, x0_max, grid_resolution),
+            np.linspace(x1_min, x1_max, grid_resolution),
+        )
+
+        pred_func = _check_boundary_response_method(estimator, response_method)
+        response = pred_func(np.c_[xx0.ravel(), xx1.ravel()])
```

Comment:
We will have an issue here in case that `response_method="predict"` and that the we are in classification with non-encoded labels. The response can be non-numeric values that cannot be converted into numeric values. The workaround would be to initialized a `LabelEncoder` with the `classes_` fitted attributes that are `estimator.classes_`. I think that it would be enough

Reply:
There is still this issue. If I remember I triggered the bug while using the class.

Reply:
Fixed in [`b864194` (#16061)](https://github.com/scikit-learn/scikit-learn/pull/16061/commits/b864194b551173689aa6ad07341da01a4f6b54f7). It uses `LabelEncoder` as you suggested. `LabelEncoder` is a bit more robust just in case the classes were integer but did not start with 0. i.e. `classes_ = [1, 3]`.

There is already a test for the string use case here: https://github.com/thomasjpfan/scikit-learn/blob/b864194b551173689aa6ad07341da01a4f6b54f7/sklearn/inspection/_plot/tests/test_boundary_decision_display.py#L306